packages/components/src/components/editor/condition-builder/ConditionBuilder.tsx

```diff
@@ -0,0 +1,72 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { ConditionGroup } from './ConditionGroup';
+import { ConditionBuilderProvider, useConditionBuilderContext, type ConditionBuilderProviderProps } from './ConditionBuilderContext';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Button } from '@/components/common/button/button';
+import { useEffect, useState } from 'react';
+import type { Browser } from '@/components/editor/browser/browser';
+
+interface ConditionBuilderProps {
+  onChange: (value: string) => void;
+}
+
+type ConditionBuilderHookProps = Omit<ConditionBuilderProviderProps, 'children'>;
+
+const conditionModes = [
+  { label: 'Basic Condition', value: 'basic-condition' },
+  { label: 'Nested Condition', value: 'nested-condition' },
+  { label: 'Always True', value: 'always-true' },
+  { label: 'Always False', value: 'always-false' }
+];
+export type ConditionMode = (typeof conditionModes)[number]['value'];
+
+export const useConditionBuilder = (props: ConditionBuilderHookProps): Browser => {
+  const [value, setValue] = useState<string>();
+
+  return {
+    name: 'Condition',
+    browser: (
+      <ConditionBuilderProvider {...props}>
+        <ConditionBuilder onChange={setValue} />
+      </ConditionBuilderProvider>
+    ),
+    applyModifier: () => {
+      return { value: value || '' };
+    },
+    infoProvider: () => <p>{value}</p>,
+    icon: IvyIcons.Process
+  };
+};
+
+export const ConditionBuilder = ({ onChange }: ConditionBuilderProps) => {
+  const { addConditionGroup, generateConditionString, conditionGroups, conditionMode, setConditionMode } = useConditionBuilderContext();
+
+  useEffect(() => {
+    onChange(generateConditionString());
+  }, [generateConditionString, onChange]);
```

Comment:
why is this needed?

Reply:
This is needed to ensure that the Info-Collapsible displays the current condition at all times. The info is defined outside the builder and can be edited via onChange.

Reply:
maybe I'm wrong, but this looks like this will be executed a lot... At least multiple times per change in the builder.
Because when does the useEffect here know that something has changed inside?